test/unit_test/static/quantity_point_kind_test.cpp

```diff
@@ -305,7 +306,7 @@ static_assert(!constructible_or_convertible_from<nth_apple<one, int>>(quantity_p
 static_assert(!constructible_or_convertible_from<nth_apple<one, int>>(quantity_point(dimensionless<percent, double>(1))));
 static_assert(!constructible_or_convertible_from<nth_apple<one, double>>(quantity_point(1.0 * s)));
 
-static_assert(construct_from_only<quantity_point_kind<time_point_kind, second, int>>(sys_seconds{42s}).relative().common() == 42 * s);
+static_assert(construct_from_only<quantity_point_kind<time_point_kind, second, int, sys_clock_origin>>(sys_seconds{42s}).relative().common() == 42 * s);
```

Comment:
I'm sure the tests will make this evident, but there's no reason for a `quantity_point` with the default origin to not be constructible from a `std::chrono::time_point`.

Reply:
Actually, a `quantity_point` with default origin should not be constructible from a `std::chrono::time_point`, as that is not one type but a template on the `Clock` (+ the duration type). That `Clock` type performs the role of the `point_origin` here. You can verify e.g. from [`operator-`](https://en.cppreference.com/w/cpp/chrono/time_point/operator_arith2) that indeed, `std::chrono::time_point`s referring to different `Clock`s are not compatible with each other.

Of course however, we should be able to deduce the correct _`point_origin` specific to the `Clock`_ (not the default `point_origin`), when constructing from a `std::chrono::time_point` using CTAD. That should already be in the PR, which of course should be tested (modulo GCC ICE). I would assume that the GCC ICE is the reason why these tests do not do CTAD? Maybe we should introduce such tests and just leave them out using preprocessor selection for affected compilers.

Reply:
I think you're right. After all, a `units::quantity_point` doesn't carry a `Clock`. Before this PR, I'm pretty sure you could do `static_assert(quantity_point{sys_seconds{0}} == quantity_point{after_christ_second{0}});` and have that assertion pass, even though `sys_seconds{0}` is 1970 A.C. and `after_christ_second{0}` is 0 A.C. With this PR, it'd become a compile-time error.

As you can glean from my example here, you _can_ do some kind of CTAD in GCC for types in this library (look for "CTAD" in the tests). Although not generally. as in the GCC bug I pointed out in other comment. For the case you mention, you should be able to use CTAD.

Reply:
I think that `static_assert(quantity_point{sys_seconds{0}} == quantity_point{after_christ_second{0}});` should be allowed. There seems to be no good excuse to not compare quantity points with different orgins?

Reply:
Probably. It's just that in `master`, the assertion passes. With this PR, it will fail at compile time. Which is better, as `master` gives the wrong result.

In the future, that can be made to work again. But you have to decide how to handle the comparison. For example, `quantity` comparison will convert both arguments to the common type, which should result in a lossless conversion that preserves the value and thus gives the correct result. But there's no precedent as to how to do that with quantity points and their origins.

For construction, there's only one way. Convert the argument to the type, adjusting it's value by the difference of origins. While it sounds simple, its solution could be involved, like `quantity_cast` is. Otherwise `quantity_point<dim_time, second, int8_t>{sys_seconds{1729}}` could overflow.